core/indexer/factory/indexerFactory_test.go

```diff
@@ -0,0 +1,160 @@
+
+	return &ArgsIndexerFactory{
+		Enabled:                  true,
+		IndexerCacheSize:         100,
+		ShardID:                  0,
+		Url:                      ts.URL,
+		UserName:                 "",
+		Password:                 "",
+		Marshalizer:              &mock.MarshalizerMock{},
+		Hasher:                   &mock.HasherMock{},
+		EpochStartNotifier:       &mock.EpochStartNotifierStub{},
+		NodesCoordinator:         &mock.NodesCoordinatorMock{},
+		AddressPubkeyConverter:   &mock.PubkeyConverterMock{},
+		ValidatorPubkeyConverter: &mock.PubkeyConverterMock{},
+		IndexTemplates:           nil,
+		IndexPolicies:            nil,
+		Options:                  &indexer.Options{},
+	}
+}
+
+func TestNewIndexerFactory(t *testing.T) {
+	tests := []struct {
+		name     string
+		argsFunc func() *ArgsIndexerFactory
+		exError  error
+	}{
+		{
+			name: "InvalidCacheSize",
+			argsFunc: func() *ArgsIndexerFactory {
+				args := createMockIndexerFactoryArgs()
+				args.IndexerCacheSize = -1
+				return args
+			},
+			exError: indexer.ErrNegativeCacheSize,
+		},
+		{
+			name: "NilAddressPubkeyConverter",
+			argsFunc: func() *ArgsIndexerFactory {
+				args := createMockIndexerFactoryArgs()
+				args.AddressPubkeyConverter = nil
+				return args
+			},
+			exError: indexer.ErrNilPubkeyConverter,
+		},
+		{
+			name: "NilValidatorPubkeyConverter",
+			argsFunc: func() *ArgsIndexerFactory {
+				args := createMockIndexerFactoryArgs()
+				args.ValidatorPubkeyConverter = nil
+				return args
+			},
+			exError: indexer.ErrNilPubkeyConverter,
+		},
+		{
+			name: "NilMarshalizer",
+			argsFunc: func() *ArgsIndexerFactory {
+				args := createMockIndexerFactoryArgs()
+				args.Marshalizer = nil
+				return args
+			},
+			exError: core.ErrNilMarshalizer,
+		},
+		{
+			name: "NilHasher",
+			argsFunc: func() *ArgsIndexerFactory {
+				args := createMockIndexerFactoryArgs()
+				args.Hasher = nil
+				return args
+			},
+			exError: core.ErrNilHasher,
+		},
+		{
+			name: "NilNodesCoordinator",
+			argsFunc: func() *ArgsIndexerFactory {
+				args := createMockIndexerFactoryArgs()
+				args.NodesCoordinator = nil
+				return args
+			},
+			exError: core.ErrNilNodesCoordinator,
+		},
+		{
+			name: "NilEpochStartNotifier",
+			argsFunc: func() *ArgsIndexerFactory {
+				args := createMockIndexerFactoryArgs()
+				args.EpochStartNotifier = nil
+				return args
+			},
+			exError: core.ErrNilEpochStartNotifier,
+		},
+		{
+			name: "EmptyUrl",
+			argsFunc: func() *ArgsIndexerFactory {
+				args := createMockIndexerFactoryArgs()
+				args.Url = ""
+				return args
+			},
+			exError: core.ErrNilUrl,
+		},
+		{
+			name: "All arguments ok",
+			argsFunc: func() *ArgsIndexerFactory {
+				return createMockIndexerFactoryArgs()
+			},
+			exError: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.exError == nil {
+				print("a")
+			}
+			_, err := NewIndexer(tt.argsFunc())
+			require.Equal(t, tt.exError, err)
```

Comment:
To work with the above mentioned comment, this can be changed to `require.True(t, errors.Is(err, tt.exError))`

Reply:
done.